src/records.jl

```diff
@@ -26,49 +28,91 @@ Simply wraps the value `x` in a `Nullable` and sticks that in an `Attribute` wit
 empty `Function`.
 """
 Attribute(x::T) where {T} = Attribute{T}(() -> x)
+Attribute{T}(x::T) where {T} = Attribute{T}(() -> x)
 
 """
     get(attr::Attribute{T}) -> T
 
 Run set `attr.x` to the output of `attr.f` if `attr.x` is not already set.
 We then return the value stored in `attr.x`
 """
-function Base.get(attr::Attribute)
+function Base.get(attr::Attribute{T}) where T
     if isnull(attr.x)
-        attr.x = Nullable(attr.f())
+        attr.x = Nullable{T}(attr.f())
     end
 
-    return get(attr.x)
+    return get(attr.x)::T
 end
 
+hasfield(T::DataType, name::Symbol) = Base.fieldindex(T, name, false) > 0
```

Comment:
Might be worth submitting this as a PR to Base. I'd definitely support that.

Reply:
https://github.com/JuliaLang/julia/pull/27582